src/Workspaces/Core/Portable/FindSymbols/StreamingProgressCollector.cs

```diff
@@ -69,15 +68,15 @@ public ValueTask OnDefinitionFoundAsync(SymbolGroup group, CancellationToken can
         }
     }
 
-    public ValueTask OnReferencesFoundAsync(
-        ImmutableArray<(SymbolGroup group, ISymbol symbol, ReferenceLocation location)> references, CancellationToken cancellationToken)
+    public async ValueTask OnReferencesFoundAsync(
+        IAsyncEnumerable<(SymbolGroup group, ISymbol symbol, ReferenceLocation location)> references, CancellationToken cancellationToken)
     {
-        lock (_gate)
+        await foreach (var (_, definition, location) in references)
         {
-            foreach (var (_, definition, location) in references)
+            lock (_gate)
```

Comment:
>lock (_gate) [](http://example.com/codeflow?start=12&length=12)

nit: this takes the lock more now, right? No contention concerns?

Reply:
it takes it as much as before.   just the loop is here, instead of receiving the individual messages and taking the lock.  i'm ok with this as i haven't seen it show up.  if it does show up, we can adjust.